drivers/nxp/ddr/nxp-ddr/ddr.c

```diff
@@ -0,0 +1,903 @@
+/*
+ * Copyright 2018-2019 NXP
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include <platform_def.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <common/debug.h>
+#include <errno.h>
+#include <string.h>
+#include <io.h>
+#include "ddr.h"
+#include <i2c.h>
+
+extern uint64_t get_timer_val(uint64_t start);
+
+struct dynamic_odt {
+	unsigned int odt_rd_cfg;
+	unsigned int odt_wr_cfg;
+	unsigned int odt_rtt_norm;
+	unsigned int odt_rtt_wr;
+};
+
+#ifndef CONFIG_STATIC_DDR
+#if defined(PHY_GEN2_FW_IMAGE_BUFFER) && !defined(NXP_DDR_PHY_GEN2)
+#error Missing NXP_DDR_PHY_GEN2
+#endif
+#ifdef NXP_DDR_PHY_GEN2
```

Comment:
The options should be documented somewhere. At least create a list of the available options, even if they don't describe much of what they do. I understand that most DDR drivers are mostly magic code with magic constants.